condition/status.go

```diff
@@ -163,3 +163,83 @@ func CheckConditionInProgress(conditionID, facilityCode, kvBucket string, js nat
 		return Indeterminate, newErrQueryStatus(errWorker, kvBucket, lookupKey, sv.WorkerID)
 	}
 }
+
+func NewTaskStatusRecord(s string) StatusRecord {
+	sr := StatusRecord{}
+	if s == "" {
+		return sr
+	}
+
+	sr.Append(s)
+
+	return sr
+}
+
+var (
+	ErrInvalidStatusRecord = errors.New("invalid status record")
+)
+
+// StatusRecord holds status information for a Condition
+type StatusRecord struct {
+	StatusMsgs []StatusMsg `json:"records"`
+}
+
+// StatusMsg is a single record within the StatusRecord
+type StatusMsg struct {
+	Timestamp time.Time `json:"ts,omitempty"`
+	Msg       string    `json:"msg,omitempty"`
+}
+
+func (sr *StatusRecord) Append(s string) {
+	if s == "" {
+		return
+	}
+
+	for _, r := range sr.StatusMsgs {
```

Comment:
We might be better served here by a simpler deduplication, comparing only to the last entry in the status messages. For example, we might have 2 discrete power-cycle steps in a condition separated by a firmware install. This seems like it would deduplicate the status messages of the second power-cycle. Thoughts?

Reply:
Thats a good point, since this is just moving the code from Flasher to rivets, I'd like to deal with that optimisation separately. I'll create a follow up issue for it.